### PR TITLE
Fixes issues with anonymity of metrics

### PR DESCRIPTION
**Pull request by itay-grudev**
This is due to the known attacks against the SHA-1 algorithm. Should improove the user anonymity and encourage users not to opt-out of the atom-metrics feature.

### CODE REVIEW

**Comment by itay-grudev** [comment]
After doing some research I came to the conclusion, that hasing the MAC address DOES NOT guaruantee anonymity no matter what hash function is used. The fallback solution that was used in case the computer cannot obtain the computer MAC address, namely the v4 UUID which is a 128bit string composed of 122bit random number is much more secure since there is no known way of pointing back to the device that created it.

**Comment by itay-grudev** [comment]
After doing some research I came to the conclusion, that hasing the MAC address DOES NOT guaruantee anonymity no matter what hash function is used. The fallback solution that was used in case the computer cannot obtain the computer MAC address, namely the v4 UUID which is a 128bit string composed of 122bit random number is much more secure since there is no known way of pointing back to the device that created it.

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Metrics package.
 
 ### Collected Data
 
-* A unique identifier that is generated by computing the [SHA-1][SHA1] of the
+* A unique identifier that is generated by computing the [SHA-256][SHA-256] of the
   machine's [MAC address][MAC].
```
Comment (50Wliu) [inline README.md]: The README needs to be updated as well to reflect that the MAC address is no longer used.

Comment (itay-grudev) [inline README.md]: That is correct. Give me a scond!

**Comment by itay-grudev** [comment]
I don't understand why you still haven't accepted the changes. The current code directly violates user's annonimity and submits their MAC addresses to a third party, in this case Google. This is highly immoral and unethical, especially from something that is Open Source when one of the reason we created Open Source is to protect users freedom.

File: README.md [hunk 2]
```diff
@@ -9,8 +9,7 @@ Metrics package.
 
 ### Collected Data
 
-* A unique identifier that is generated by computing the [SHA-1][SHA1] of the
-  machine's [MAC address][MAC].
+* A unique UUID v4 random identifier is generated according to [RFC4122][RFC4122]
```
Comment (50Wliu) [inline README.md]: :memo: missing a period here.

**Comment by 50Wliu** [comment]
Looks good to me.  cc @atom/feedback

File: README.md (hunk 2 above)
Comment (itay-grudev) [inline README.md]: The other entries in the list don't have periods as well. Wouldn't it look rather odd?

Comment (50Wliu) [inline README.md]: Ah, you're totally right.  In that case, the missing period should be fine.

**Comment by maxbrunsfeld** [comment]
> After doing some research I came to the conclusion, that hashing the MAC address DOES NOT guarantee anonymity no matter what hash function is used.

Could you exactly what research brought you to this conclusion? I'm curious, and I'd like to understand so that I can help consider the tradeoffs here. Hashing the MAC address is useful to us because it provides a user id that persists even if the user removes their `config.cson`, which is common.

**Comment by itay-grudev** [comment]
It is very simple. A MAC address is a 48bit number. From those 48 bits 24 are vendor prefixes and the other 24 bits are a unique address within the vendor space. Since there are only about 20,000 allocated vendor prefixes that makes about 2<sup>38</sup> valid assigned MAC addresses. Modern consumer hardware can calculate 2<sup>30</sup> hashes per second. Thus you would need only 2<sup>8</sup> seconds to go through all the possible MAC addresses in the worst case. That's 4 minutes!

Mayer, Jonathan. "Questionable Crypto in Retail Analytics." Web log post. Web Policy. N.p., n.d. Web. 28 Sept. 2015.

**Comment by lee-dohm** [comment]
I suspect the main concern is that the MAC address space is 48 bits while the SHA-1 space is 160 bits. This means that the majority if not all or nearly all of the possible MAC addresses hash to separate buckets, making it ripe for a dictionary attack. 

http://crypto.stackexchange.com/questions/19974/

**Comment by maxbrunsfeld** [comment]
> It is very simple.

Great; thanks for explaining.

**Comment by maxbrunsfeld** [comment]
This will be fixed in v1.1.0. Thanks @itay-grudev.

**Comment by itay-grudev** [comment]
Thanks for taking this into consideration and merging it.

**Comment by jite** [comment]
I tried to explain this a year ago, luckily @itay-grudev did a better job and problem is now fixed ;)

**Pull request merged**